src/android/Capture.java

```diff
@@ -234,6 +246,17 @@ private void captureAudio(Request req) {
           try {
               Intent intent = new Intent(android.provider.MediaStore.Audio.Media.RECORD_SOUND_ACTION);
 
+              String timeStamp = new SimpleDateFormat("yyyyMMddHHmmssSSS").format(new Date());
+              String fileName = "AUDIO_" + timeStamp + ".wav";
+              File audio = new File(getTempDirectoryPath(), fileName);
```

Comment:
(for all 3 capture types)

I would drop the use of `getTempDirectoryPath` to not write in app temp dir, because this path is cleaned up by the system while such captured audio/image/video should be available afterward in the device media dirs (correct [Environment.DIRECTORY_..](https://developer.android.com/reference/android/os/Environment#DIRECTORY_PICTURES)) as a common practice.

See https://developer.android.com/training/camera/photobasics#TaskPath
- can there be 3 isolated methods like the suggested `createImageFile()`? and not inlined like here
- use `File.createTempFile`
- with specific values and correct Environment.DIRECTORY_.. for each of our 3 types
- and possibly more `<external-files-path` entries to match that

Reply:
The Android training link I shared gives example of using `getExternalFilesDir` (with corresponding _paths.xml_) but this writes to the app private dirs and thus requires to "Add the photo to a gallery" to be similar to the auto-indexing of previous Android and iOS mediacapture code!
> Note: If you saved your photo to the directory provided by getExternalFilesDir(), the media scanner cannot access the files because they are private to your app.

Do you agree? Or to which dirs will we agree to write _by default_ here?

Any way to capture and write all 3 types to **public** dirs?

Reply:
https://developer.android.com/guide/topics/media/camera.html#saving-media
gives example of `mediaStorageDir` defined using `Environment.getExternalStoragePublicDirectory`

but [deprecated](https://developer.android.com/reference/android/os/Environment#getExternalStoragePublicDirectory(java.lang.String)) since API 29: what to use? are this [SO](https://stackoverflow.com/questions/57116335/environment-getexternalstoragedirectory-deprecated-in-api-level-29-java) or other correct? with pre and post Android Q code? through ContentResolver? through ACTION_CREATE_DOCUMENT? ...